conjure/unionwriter_test.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2022 Palantir Technologies. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package conjure
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/dave/jennifer/jen"
+	"github.com/palantir/conjure-go/v6/conjure/types"
+	"github.com/stretchr/testify/assert"
+)
+
+var testUnionType = &types.UnionType{
+	Name: "MyUnion",
+	Fields: []*types.Field{
+		{
+			Name: "stringVal", Type: types.String{},
+		},
+		{
+			Name: "boolVal", Type: types.Boolean{},
+		},
+	},
+}
+
+func TestUnionWriter_unionVisitorInterfaceT(t *testing.T) {
```

Comment:
This is the file of interest in terms of seeing the code the new methods generate - once we agree on what we want these to look like, I will go ahead and move from draft to formal PR where the tested methods are actually invoked.

Reply:
Would you mind adding some integration tests or a branch somewhere that has an example of using the generated code? I think that being able to see the actual generated code would be helpful in doing the review.

Reply:
Added a basic example of usage here: https://github.com/jamesross3/conjure-go/blob/jr/union-generics-example-usage/conjure-api/conjure/spec/unions.conjure_test.go

The code that actually gets generated is now present in the latest commit ( in the file `conjure-api/conjure/spec/unions.conjure.go`)

Reply:
Thanks, seeing the example was very helpful!